Makefile

```diff
@@ -26,7 +26,8 @@ ifeq ($(UNAME_S),Darwin)
 	MOBY_DOCKER=/Applications/Docker.app/Contents/Resources/bin/docker
 endif
 
-TEST_FLAGS?=
+TEST_COVERAGE_FLAGS = -race -coverprofile=coverage.out -covermode=atomic
```

Comment:
@nicksieger 
Note that it adds "-race" to detect data races.
The changes in the tests are related to this.